src/discounts/handlers.ts

```diff
@@ -117,19 +126,21 @@ export const getChannelsVariables = (
     id,
     input: {
       addChannels:
-        formData.channelListings?.map(channel => ({
-          channelId: channel.id,
-          discountValue:
-            formData.discountType.toString() === "SHIPPING"
-              ? 100
-              : channel.discountValue,
-          minAmountSpent:
-            formData.requirementsPicker === RequirementsPicker.NONE
-              ? null
-              : formData.requirementsPicker === RequirementsPicker.ITEM
-              ? 0
-              : channel.minSpent,
-        })) || [],
+        formData.channelListings
+          ?.map(channel => ({
+            channelId: channel.id,
+            discountValue:
+              formData.discountType.toString() === "SHIPPING"
+                ? 100
+                : channel.discountValue,
+            minAmountSpent:
+              formData.requirementsPicker === RequirementsPicker.NONE
+                ? null
+                : formData.requirementsPicker === RequirementsPicker.ITEM
+                ? 0
+                : channel.minSpent,
+          }))
+          .filter(channel => !!channel.discountValue) || [],
```

Comment:
suggestion: would be nice to create a dedicated function for it, `mapFormChannelListingToInput` or something. Also, the map function can be split into smaller chunks for readability.

Reply:
example:

```js
const mapDiscoutValue = (discountType) => (channel) => ({
  ...channel,
  discountValue: discountType === "SHIPPING" ? 100 : channel.discountValue,
})

const getRequirementsPicker = (requirementsPicker, minSpent) => {
  if (requirementsPicker === RequirementsPicker.NONE) {
    return null
  }

  if (requirementsPicker === RequirementsPicker.ITEM) {
    return 0
  }

  return minSpent
}

const mapMinAmountSpent = (requirementsPicker) => (channel) => ({
  ...channel,
  minAmountSpent: getRequirementsPicker(requirementsPicker, channel.minSpent)
})

const filterNotDiscounted = channel => !!channel.discountValue

export const createInputByFormData = (formData) => {
  return formData
    .map(mapDiscoutValue(formData.discountType.toString()))
    .map(mapMinAmountSpent(formData.requirementsPicker))
    .filter(filterNotDiscounted)
}
```

Reply:
I've changed it to similar, but with one map function